cdap-app-fabric/src/main/java/co/cask/cdap/app/preview/PreviewServer.java

```diff
@@ -0,0 +1,287 @@
+                                         final ArtifactStore artifactStore,
+                                         final AuthorizerInstantiator authorizerInstantiator,
+                                         final StreamAdmin streamAdmin,
+                                         final StreamConsumerFactory streamConsumerFactory,
+                                         final StreamCoordinatorClient streamCoordinatorClient,
+                                         final PrivilegesManager privilegesManager,
+                                         final AuthorizationEnforcer authorizationEnforcer) {
+    // create cConf and hConf and set appropriate settings for preview
+    CConfiguration previewcConf = CConfiguration.copy(cConf);
+    previewcConf.set(Constants.CFG_LOCAL_DATA_DIR, previewcConf.get(Constants.CFG_LOCAL_PREVIEW_DIR));
+    previewcConf.set(Constants.Dataset.DATA_DIR, previewcConf.get(Constants.CFG_LOCAL_PREVIEW_DIR));
+    Configuration previewhConf = new Configuration(hConf);
+    File localPreviewDir = new File(cConf.get(Constants.CFG_LOCAL_PREVIEW_DIR));
+    previewcConf.set(Constants.CFG_LOCAL_DATA_DIR, localPreviewDir.getAbsolutePath());
+    previewcConf.setIfUnset(Constants.CFG_DATA_LEVELDB_DIR, "data/preview");
+    return Guice.createInjector(
+      new ConfigModule(previewcConf, previewhConf),
+      new IOModule(),
+      new AuthenticationContextModules().getMasterModule(),
+      // todo : figure if we should we share secure store modules to allow reading the secure data during preview
+      new SecurityModules().getStandaloneModules(),
+      new SecureStoreModules().getStandaloneModules(),
+      new PreviewDiscoveryRuntimeModule(discoveryService),
+      new LocationRuntimeModule().getStandaloneModules(),
+      new ConfigStoreModule().getStandaloneModule(),
+      new PreviewServerModule(),
+      new ProgramRunnerRuntimeModule().getStandaloneModules(),
+      new PreviewModules().getDataFabricModule(transactionManager),
+      new PreviewModules().getDataSetsModule(datasetFramework),
+      new DataSetServiceModules().getStandaloneModules(),
+      new MetricsClientRuntimeModule().getStandaloneModules(),
+      new LoggingModules().getStandaloneModules(),
+      new NamespaceStoreModule().getStandaloneModules(),
+      new RemoteSystemOperationsServiceModule(),
+      new AbstractModule() {
+        @Override
+        protected void configure() {
+          bind(ArtifactRepository.class).toInstance(artifactRepository);
+          bind(ArtifactStore.class).toInstance(artifactStore);
+          bind(AuthorizerInstantiator.class).toInstance(authorizerInstantiator);
+          bind(AuthorizationEnforcer.class).toInstance(authorizationEnforcer);
+          bind(PrivilegesManager.class).toInstance(privilegesManager);
+          bind(StreamAdmin.class).toInstance(streamAdmin);
+          bind(StreamConsumerFactory.class).toInstance(streamConsumerFactory);
```

Comment:
It's for deleting app so that it cleans up the stream consumer states (used by Flow). However, in preview case, I believe the stream consumer states shouldn't be the regular one, but rather a preview specific one (e.g. using `InMemoryStreamConsumerFactory` instead of the regular one used by CDAP).